packages/core/src/service/services/order.service.ts

```diff
@@ -1328,8 +1328,25 @@ export class OrderService {
             promotions,
             updatedOrderLine ? [updatedOrderLine] : [],
         );
+        // await this.connection.getRepository(ctx, OrderItem).save(updatedItems, { reload: false });
+        await this.connection
+            .getRepository(ctx, OrderItem)
+            .createQueryBuilder()
+            .insert()
+            .values(updatedItems)
+            .orUpdate({
+                conflict_target: ['id'],
+                overwrite: [
```

Comment:
For maintainability it would be good to extract this array into a var with type `keyof OrderItem` to make it slightly more resistant to refactoring errors.